### PR TITLE
⚡️⚡️⚡️ Lightning Network support 

### LINKED ISSUE #142: Expose -generate flag of bitcoin-cli to `nigiri rpc`
I’m trying to use bitcoin-cli’s `generate` with Nigiri, but to no avail:

```
❯ nigiri rpc generate 10
error code: -1
error message:
generate
has been replaced by the -generate cli option. Refer to -help for more information.
exit status 1

❯ nigiri rpc -generate 10
Incorrect Usage: flag provided but not defined: -generate

NAME:
   nigiri rpc - invoke bitcoin-cli or elements-cli

USAGE:
   nigiri rpc [command options] [arguments...]

OPTIONS:
   --liquid           enable liquid (default: false)
   --rpcwallet value  rpcwallet to be used for node JSONRPC commands
   --help, -h         show help (default: false)

[nigiri] flag provided but not defined: -generate
```

This is what I expect would happen:

```
❯ bitcoin-cli -regtest -generate 10
{
  "address": "bcrt1qpcul9qzkketgqseehq06s75mxcttpv9q4xfpdt",
  "blocks": [
    "7e00a57d010ca30c7e71e797dfddb699abd3537f6dfb1b30d817eef4ee027b00",
    "33c94b24a193edfa8968242df3dc126003bdd3bfa161410988a563537b9bef25",
    "370d4e1f6ac122551f636cc106ce3110d1fd5c37e96c23b952649efdb6186491",
    "5db7c34629421a839c968a3bb10baa44ba7ce98e093146d92eae6f6ee1c79ad6",
    "411d51bed5809c5433e0cca66ee1827f7ce8a172e128a229ad10c68c2fecab70",
    "3c1d97b863da244c707a0cc6aecc6ca4b2dccbc2092c14fa8e68c776635c9c7a",
    "547d72889dd95f03fc0e01766eb74620a3deac7ed40e624ba77dd5a802d3992a",
    "01d4cabae7d491136a52a288b89c8bb1bf080788510f0931335188e92c97140b",
    "7b28cf723ed6f27c2e38cc4fa675dd2a4ea5df23d1148af10f21a97ed2137c0d",
    "3f0a17786b88b141010ff358b2ccefbfdc841928161e09d615dc6661da6de771"
  ]
}
```

### PR DESCRIPTION
**Pull request by tiero**
This adds support of a much requested feature: ⚡️ Lightning Network support in Nigiri﻿ 🎉

## How to test?

1. Go here and grab the binary for your os/arch combo
https://github.com/tiero/nigiri/releases/tag/v0.4.0-ln.0
2. `sudo chmod a+x nigiri-x-y`
3.  `rm -rf ~/Library/Application\ Support/Nigiri`
4. `alias nigiri="./nigiri-x-y"`
5. `nigiri start --ln`

```sh
# start LND and CLN at same time
nigiri start --ln 
# or can be run with liquid together
nigiri start --liquid --ln

# get an address from LND
nigiri lnd newaddress p2wkh

# get an address from Core Lightning
nigiri cln newaddr

# fund LND one-shot
nigiri faucet `nigiri lnd newaddress p2wkh | jq -r .address`

# fund Core Lightning one-shot
nigiri faucet `nigiri cln newaddr | jq -r .bech32` 0.1
```

## Is ready fro review?

- [x] add one instances
 of LND with `noseedbackup` and one of Core Lightning.
- [x] Wrap `lncli` & `lighting-cli` command-line interfaces under `nigiri lnd` and/or `nigiri cln`
- [x] Add Core Lighting service 
- [x] Allow to fund cln & lnd one command `nigiri faucet cln` or `nigiri faucet lnd` 
- [x] Update README documentation

This PR is ready to review/merge, but worth mentioning other "nice to have tasks" that need chopsticks changes

- [ ] Automate node funding and channel creation between nodes
- [ ] Intercept somehow on-chain transaction from node and trigger block mining 

**Bonus**
This also closes #142

### CODE REVIEW

**Comment by tiero** [comment]
Ready for review/testing @altafan

File: README.md [hunk 1]
```diff
@@ -41,128 +41,70 @@ $ nigiri start --liquid
...
 
 ```bash
 # Bitcoin
-$ nigiri faucet bcrt1qsl4j5je4gu3ecjle8lckl3u8yywh8rff6xxk2r
+$ nigiri faucet <bitcoin_address>
+## Fund the Core Lightning node
+$ nigiri faucet cln 0.01
+
+## Fund the LND node
+$ nigiri faucet lnd 0.01
```
Comment (altafan) [inline README.md:95]: I'm following the readme like a noob and these 2 instructions are misleading.. what do you mean with `cln` and `lnd`? how to generate bitcoin addresses from those services?

File: README.md [hunk 2]
```diff
@@ -41,128 +41,70 @@ $ nigiri start --liquid
...
 ```
 Use the `--delete` flag to not just stop Docker containers but also to remove them and delete the config file and any new data written in volumes.
 
-* Generate and send bitcoin to given address
+### Generate and send bitcoin to given address
 
 ```bash
 # Bitcoin
-$ nigiri faucet bcrt1qsl4j5je4gu3ecjle8lckl3u8yywh8rff6xxk2r
+$ nigiri faucet <bitcoin_address>
```
Comment (altafan) [inline README.md:90]: I get an error on this:
```bash
$ nigiri faucet bcrt1q5vye7el7at30g6g7lchapxhef8efjds2rzz3nz
# [nigiri] Method sendtoaddress failed with error: Invalid Bitcoin address:
```

Comment (altafan) [reply README.md:90]: seems like the address is an empty string when passed down to the bitcoin service

File: README.md [hunk 3]
```diff
@@ -41,128 +41,70 @@ $ nigiri start --liquid
...
+## Fund the Core Lightning node
+$ nigiri faucet cln 0.01
+
+## Fund the LND node
+$ nigiri faucet lnd 0.01
 
 # Elements
-$ nigiri faucet --liquid el1qqwwx9gyrcrjrhgnrnjq9dq9t4hykmr6ela46ej63dnkdkcg8veadrvg5p0xg0zd6j3aug74cv9m4cf4jslwdqnha2w2nsg9x3
+$ nigiri faucet --liquid <liquid_address>
```
Comment (altafan) [inline README.md:98]: Same here:

```bash
$ nigiri faucet --liquid el1qq2cs67karwfgzmt02d2ps2dnnvlgtyju8r9d7gwckzmyx45v9l8xw0zsxrtrrqhq50pj0d096hn0pphnfa25je8e9rfcm8glp
# [nigiri] Method sendtoaddress failed with error: Invalid Bitcoin address:
```

**Review by altafan — COMMENTED** [review]
Besides the problem with the `faucet` cmd, everything LGTM.

I followed the instructions of the readme and I was able to open a channel between the 2 ln nodes. I made sure of this by checking the logs of the 2 services.

~~Feature request: it would be nice to access the commands of the `ln-cli` and `cln-cli` via nigiri's `rpc` cmd like possible for _bitcoin_ and _liquid_ services.~~

**Review by altafan — APPROVED** [review]
tACK.

**Pull request merged**